src/machine-learning/machine-learning.ts

```diff
@@ -252,13 +295,73 @@ export class ModelOptions {
   public displayName?: string;
   public tags?: string[];
 
-  public tfliteModel?: { gcsTFLiteUri: string; };
+  public tfliteModel?: { gcsTfliteUri: string; };
+}
 
-  protected toJSON(forUpload?: boolean): object {
-    throw new Error('NotImplemented');
+async function convertOptionstoContent(
+    options: ModelOptions, forUpload?: boolean,
+    storageClient?: StorageClient): Promise<object> {
+  const modelContent = deepCopy(options);
+  if (forUpload && modelContent.tfliteModel?.gcsTfliteUri) {
+    if (!storageClient) {
+      throw new FirebaseMachineLearningError(
+        'invalid-argument',
+        'Must specify storage client if forUpload and gcs Uri are specified.',
+      );
+    }
+    modelContent.tfliteModel.gcsTfliteUri = await signUrl(modelContent.tfliteModel.gcsTfliteUri, storageClient!);
+  }
+  return modelContent;
+}
+
+async function signUrl(unsignedUrl: string, storageClient: StorageClient): Promise<string> {
+  const MINUTES = 60 * 1000; // A minute in milliseconds.
+  const URL_VALID_DURATION = 10 * MINUTES;
+
+  const gcsRegex = /^gs:\/\/([a-z0-9_.-]{3,63})\/(.+)$/;
+  const matches = gcsRegex.exec(unsignedUrl);
+  if (!matches) {
+    throw new FirebaseMachineLearningError(
+      'invalid-argument',
+      `Invalid unsigned url: ${unsignedUrl}`);
+  }
+  const bucketName = matches[1];
+  const blobName = matches[2];
+  const bucket = storageClient.bucket(bucketName);
+  const blob = bucket.file(blobName);
+
+  try {
```

Comment:
Try catch only works if you `await` the promise. Drop try-catch and also drop `async` in method signature. You need to handle any errors in a separate `.catch()` section.

Reply:
done